view/jpa/src/main/kotlin/io/holunda/polyflow/view/jpa/data/ConverterExt.kt

```diff
@@ -192,3 +192,26 @@ fun DataEntryDeletedEvent.toEntity(
 }.apply {
   this.protocol = this.protocol.addModification(this, this@toEntity.deleteModification, this@toEntity.state)
 }
+
+/**
+ * Event to entity for an anonymization, if an optional entry exists.
+ */
+fun DataEntryAnonymizedEvent.toEntity(
+  revisionValue: RevisionValue,
+  oldEntry: DataEntryEntity,
+) = oldEntry.also {
+  it.protocol.forEach { protocolEntry ->
+    if (protocolEntry.username != null && !this.excludedUsernames.contains(protocolEntry.username))
+      protocolEntry.username = this.anonymizedUsername
+  }
+  it.authorizedPrincipals =
+    it.authorizedPrincipals.filter { principal -> !principal.startsWith("USER:") }.toMutableSet()
+  it.lastModifiedDate = this.anonymizeModification.time.toInstant()
+  it.revision = if (revisionValue != RevisionValue.NO_REVISION) {
+    revisionValue.revision
+  } else {
+    it.revision
+  }
+}.apply {
+  this.protocol = this.protocol.addModification(this, this@toEntity.anonymizeModification, this.state.toState())
```

Comment:
By doing this in the end, you might end up with one user still in the data entry - the user who executed the anonymization. I think if you first add the modification and then anonymize, you will really get the enonymized data entry.

Reply:
This was a deliberate decision. I would expect the anonymization to be triggered by some technical user. I think it would be weird to anonymize the anonymization action somehow